### PR TITLE
Feature/quick pick

### PR DESCRIPTION
**Pull request by matesk8**
This PR contains the Qucik-pick component, that can be passed a string list to display as radio buttons, or add an expression using the Expression component.

### CODE REVIEW

File: src/app/components/quick-pick/quick-pick.component.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
+@import "../../../assets/sass/variables";
+@import "../../../assets/sass/_theme-variables";
```
Comment (mayacoda) [inline src/app/components/quick-pick/quick-pick.component.scss]: There is no need to import _theme-variables, this file is already imported at the end of variables.scss

File: src/app/components/quick-pick/quick-pick.component.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        this.removeExpressionInputSub();
+
+        this.expressionInputSub = newExpression
+            .filter(expression => expression !== undefined)
+            .subscribe((expression: ExpressionModel) => {
+                this.updateExpressionValue(expression);
+
+                this.formGroup.controls["expressionField"].setValue(
+                    this.expressionModel.getExpressionScript(), {
+                        onlySelf: true,
```
Comment (mayacoda) [inline src/app/components/quick-pick/quick-pick.component.ts]: Why doesn't this control propagate its state to its formGroup parent?

File: src/app/components/quick-pick/quick-pick.component.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    constructor(private expressionSidebarService: ExpressionSidebarService) {
+
+    }
+
+    ngOnInit() {
+        this.formGroup.addControl(
+            "radioButtonControl", new FormControl("")
+        );
+
+        this.subs.push(
```
Comment (mayacoda) [inline src/app/components/quick-pick/quick-pick.component.ts]: You can extend the `ComponentBase` class and assign subscriptions and other objects that should be destroyed to the `tracked` property.  If you're implementing the `ngOnDestroy` method, be sure to call `super.ngOnDestroy` inside it.

File: src/app/components/quick-pick/quick-pick.component.ts (hunk 2 above)
Comment (matesk8) [inline src/app/components/quick-pick/quick-pick.component.ts]: On line 106, I listen to the value change, inside the input field, which sets the new value to a string. I don't want to trigger that change detection if the the value was set to an expression, but I do want to update the input field's value.

Comment (mayacoda) [inline src/app/components/quick-pick/quick-pick.component.ts]: Will the parent form be adequately notified of changes in the model, that's my only concern. :)

Comment (matesk8) [inline src/app/components/quick-pick/quick-pick.component.ts]: Changed it.

File: src/app/components/quick-pick/quick-pick.component.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                    [formControl]="formGroup.controls['expressionField']">
+             </expression-input>
+            
+            <span class="col-sm-1">
+                <i class="fa fa-trash clear-icon" (click)="removeExpressionInput()"></i>
+            </span>
+        </div>
+    `
+})
+export class QuickPickComponent extends ComponentBase {
```
Comment (mayacoda) [inline src/app/components/quick-pick/quick-pick.component.ts]: This component should also be a custom form control like compact list.

**Pull request merged**